src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -136,12 +145,21 @@ uct_dc_mlx5_ep_atomic_fop(uct_dc_mlx5_ep_t *ep, int opcode, void *result, int ex
     uct_rc_iface_send_desc_t *desc;
 
     UCT_DC_MLX5_CHECK_RES(iface, ep);
-    UCT_RC_IFACE_GET_TX_ATOMIC_FETCH_DESC(&iface->super.super, &iface->super.tx.atomic_desc_mp,
-                                          desc, uct_rc_iface_atomic_handler(&iface->super.super,
-                                                                            ext, length),
-                                          result, comp);
+    UCT_RC_IFACE_GET_TX_ATOMIC_FETCH_DESC(&iface->super.super,
+                                          &iface->super.tx.atomic_desc_mp, desc,
+                                          uct_rc_iface_atomic_handler(
+                                                  &iface->super.super, ext,
+                                                  length),
+                                          result, comp,
+                                          {
+                                              uct_dc_mlx5_iface_dci_put(
+                                                      iface, ep->dci);
+                                              return UCS_ERR_NO_RESOURCE;
```

Comment:
maybe better do `goto err` and release dci there?
Here an in all similar places below